demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-        if (!ParseAndCheckCommandLine(argc, argv)) {
-            return 0;
-        }
+    size_t dims;
+    auto compare_dims = [&](const ov::Output<ov::Node>& input) { return input.get_shape().size() == dims; };
 
-        /** This vector stores paths to the processed images **/
-        std::vector<std::string> imagePaths;
-        parseInputFilesArguments(imagePaths);
-        if (imagePaths.empty()) throw std::logic_error("No suitable images were found");
-        // -----------------------------------------------------------------------------------------------------
+    dims = 4;
+    auto it = std::find_if(inputs.begin(), inputs.end(), compare_dims);
+    if (inputs.end() == it) {
+        throw std::logic_error("Can't find input with " + std::to_string(dims) + " dimensions");
+    }
 
-        // ---------------------Load inference engine------------------------------------------------
-        slog::info << *InferenceEngine::GetInferenceEngineVersion() << slog::endl;
-        InferenceEngine::Core ie;
+    dims = 2;
+    it = std::find_if(inputs.begin(), inputs.end(), compare_dims);
+    if (inputs.end() == it) {
+        throw std::logic_error("Can't find input with " + std::to_string(dims) + " dimensions");
+    }
 
-        if (!FLAGS_l.empty()) {
-            // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
-            auto extension_ptr = std::make_shared<InferenceEngine::Extension>(FLAGS_l);
-            ie.AddExtension(extension_ptr, "CPU");
-        }
-        if (!FLAGS_c.empty()) {
-            // clDNN Extensions are loaded from an .xml description and OpenCL kernel files
-            ie.SetConfig({{InferenceEngine::PluginConfigParams::KEY_CONFIG_FILE, FLAGS_c}}, "CPU");
-        }
+    it = std::find_if(outputs.begin(), outputs.end(), compare_dims);
+    if (outputs.end() == it) {
+        throw std::logic_error("Can't find output with " + std::to_string(dims) + " dimensions");
+    }
 
-        // -----------------------------------------------------------------------------------------------------
+    dims = 2;
+    it = std::find_if(outputs.begin(), outputs.end(), compare_dims);
+    if (outputs.end() == it) {
+        throw std::logic_error("Can't find output with " + std::to_string(dims) + " dimensions");
+    }
+}
 
-        // --------------------Load network (Generated xml/bin files)-------------------------------------------
+int main(int argc, char* argv[]) {
+    std::set_terminate(catcher);
 
-        /** Read network model **/
-        auto network = ie.ReadNetwork(FLAGS_m);
+    PerformanceMetrics metrics;
 
-        // add DetectionOutput layer as output so we can get detected boxes and their probabilities
-        network.addOutput(FLAGS_detection_output_name.c_str(), 0);
-        // -----------------------------------------------------------------------------------------------------
+    // Parsing and validation of input args
```

Comment:
Such comments are on a way of being removed:https://github.com/openvinotoolkit/open_model_zoo/pull/3157. While you are here, could you remove these comments?

Reply:
nothing wrong with comments, except they might look weird
 # ---------------------------- 6. Processing of the received inference results ------------------------------
this would be better
# Processing of the received inference results